core/storage/collection/gae_models.py

```diff
@@ -196,6 +196,37 @@ def get_deletion_policy():
         """
         return base_models.DELETION_POLICY.KEEP_IF_PUBLIC
 
+    @staticmethod
+    def transform_dict_to_valid(model_dict):
+        """Replace invalid fields and values in the CollectionRightsModel dict.
+
+        Some old CollectionRightsSnapshotContentModels can contain fields
+        and field values that are no longer supported and would cause
+        an exception when we would try to create CollectionRightsModel. We need
+        to remove or replace these fields and values.
+
+        Args:
+            model_dict: dict. The content of the model. Some fields and field
+                values might no longer exist in the CollectionRightsModel.
+
+        Returns:
+            dict. The content of the model. Only valid fields and values are
+            present.
+        """
+        # The status field could contain different values in history, all
+        # these values are equal to public now.
+        if ('status' in model_dict and
+                model_dict['status'] not in [constants.ACTIVITY_STATUS_PRIVATE,
```

Comment:
I think this should be "== 'publicized'"; ditto elsewhere.

Reply:
Done.